### PR TITLE
[Fix] Enable mini-batch r-gcn for CPU

### PR DESCRIPTION
**Pull request by mszarma**
## Description

Fix for enabling r-gcn training for CPU device in entity_classify_mp.py
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/rgcn/entity_classify_mp.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def evaluate(model, embed_layer, eval_loader, node_feats):
 
 @thread_wrapped_func
 def run(proc_id, n_gpus, args, devices, dataset, split, queue=None):
-    dev_id = devices[proc_id]
```
Comment (classicsong) [inline examples/pytorch/rgcn/entity_classify_mp.py:188]: For cpu runs, the input is already ['cpu']. Why we need to change here?

Comment (mszarma) [reply examples/pytorch/rgcn/entity_classify_mp.py:188]: Hi @classicsong thanks for comment.
Regarding you questions please look at line 261 - there is a problem with that condition and str vs int comparison.

File: examples/pytorch/rgcn/model.py [hunk 2]
```diff
@@ -118,13 +118,14 @@ def forward(self, node_ids, node_tids, type_ids, features):
             embeddings as the input of the next layer
         """
         tsd_ids = node_ids.to(self.node_embeds.weight.device)
-        embeds = th.empty(node_ids.shape[0], self.embed_size, device=self.dev_id)
+        device_id = self.dev_id if self.dev_id >= 0 else 'cpu'
```
Comment (classicsong) [inline examples/pytorch/rgcn/model.py:121]: We can make self.dev_id = th.device(dev_id if dev_id >= 0 else 'cpu') at __init__ function.
Then there is no need to do following changes.

File: examples/pytorch/rgcn/entity_classify_mp.py (hunk 1 above)
Comment (classicsong) [reply examples/pytorch/rgcn/entity_classify_mp.py:188]: I c.
Good point.
Can we just change the L514 to send in [-1] ?

**Review by classicsong — COMMENTED** [review]
Some suggestions.
Overall LGTM.

File: examples/pytorch/rgcn/entity_classify_mp.py (hunk 1 above)
Comment (BarclayII) [reply examples/pytorch/rgcn/entity_classify_mp.py:188]: I don't think so.  For instance `RelGraphEmbedLayer` uses `dev_id` rather than `device`.

**Pull request merged**